.github/workflows/ci.yml

```diff
@@ -0,0 +1,225 @@
+          # When updating this, the reminder to update the minimum required version of `async-await` feature in README.md.
+          - 1.37.0
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust
+        run: rustup update ${{ matrix.rust }} && rustup default ${{ matrix.rust }}
+      - run: cargo install cargo-hack
+      # remove dev-dependencies to avoid https://github.com/rust-lang/cargo/issues/4866
+      - run: cargo hack --remove-dev-deps --workspace
+      # Check no-default-features
+      - run: cargo hack build --workspace --exclude futures-test --ignore-private --no-default-features
+      # Check alloc feature
+      - run: cargo hack build --workspace --exclude futures-test --ignore-private --no-default-features --features alloc --ignore-unknown-features
+      # Check std feature
+      - run: cargo hack build --workspace --ignore-private --no-default-features --features std --ignore-unknown-features
+      # Check compat feature (futures, futures-util)
+      - run: cargo hack build -p futures -p futures-util --no-default-features --features std,io-compat
+      # Check thread-pool feature (futures, futures-executor)
+      - run: cargo hack build -p futures -p futures-executor --no-default-features --features std,thread-pool
+
+  build:
+    name: cargo +${{ matrix.rust }} build
+    strategy:
+      matrix:
+        rust:
+          # This is the minimum Rust version supported by `async-await` feature.
+          # When updating this, the reminder to update the minimum required version of `async-await` feature in README.md.
+          - 1.39.0
+          - stable
+          - beta
+          - nightly
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust
+        run: rustup update ${{ matrix.rust }} && rustup default ${{ matrix.rust }}
+      - run: cargo install cargo-hack
+      - run: cargo hack build --workspace --no-dev-deps
+      - run: cargo build --tests --features default,thread-pool,io-compat --manifest-path futures/Cargo.toml
+
+  minimal-versions:
+    name: cargo build -Zminimal-versions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust
+        run: rustup update nightly && rustup default nightly
+      - run: cargo run --manifest-path ci/remove-dev-dependencies/Cargo.toml */Cargo.toml
+      - run: cargo update -Zminimal-versions
+      - run: cargo build --workspace --all-features
+
+  thumbv6m:
+    name: cargo build --target thumbv6m-none-eabi
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust
+        run: rustup update nightly && rustup default nightly
+      - run: rustup target add thumbv6m-none-eabi
+      - run: cargo run --manifest-path ci/remove-dev-dependencies/Cargo.toml */Cargo.toml
```

Comment:
This is equivalent to `cargo hack --remove-dev-deps`. It may be okay to remove this script given that this script [takes as long to compile](https://github.com/rust-lang/futures-rs/runs/1287718190?check_suite_focus=true
) as [installing cargo-hack](https://github.com/rust-lang/futures-rs/runs/1287717540?check_suite_focus=true).

FYI @Nemo157